python/ray/util/sgd/torch/torch_runner.py

```diff
@@ -189,10 +181,13 @@ def train_epoch(self, num_steps=None, info=None):
             train_stats = self.training_operator.train_epoch(iterator, info)
 
         self.epochs += 1
-        train_stats.update(self.stats())
-        return train_stats
+        stats = {"epoch": self.epochs}
+        stats.update(train_stats)
+        if profile:
+            stats.update(profile=self.time_stats())
```

Comment:
Maybe we should make the timers themselves no-op if profiling is off, instead of just throwing the data away here.

Reply:
Done.